tests/reftests/filter-variable-resolution.test

```diff
@@ -0,0 +1,1111 @@
+N0REP0
+### : Variable definition in all filters :
+### OPAMSTRICT=0
+### <pkg:foo.1>
+opam-version: "2.0"
+### <pkg:bar.1>
+opam-version: "2.0"
+### <pkg:baz.1>
+opam-version: "2.0"
+### <pkg:undefined-smtg.1>
+opam-version: "2.0"
+### ::::::::::::::::::::::
+### :I: Global variable ::
+### ::::::::::::::::::::::
+### opam switch create global-variable --empty
+### opam var smtg=something --global
+Added '[smtg "something" "Set through 'opam var'"]' to field global-variables in global configuration
+### :I:1: patches :::
+### <pkg:patches.1:some-content>
+blabla
+pioupiou
+bloblob
+### <pkg:patches.1:to-apply.patch>
+--- a/some-content	2020-12-02 14:22:55.364620832 +0100
+++ b/some-content	2020-12-02 14:23:05.668686881 +0100
+@@ -1,3 +1,3 @@
+ blabla
+-pioupiou
++ploplop
+ bloblob
+### <pkg:patches.1:to-not-apply.patch>
+--- a/some-content	2020-12-02 14:22:55.364620832 +0100
+++ b/some-content	2020-12-02 14:23:05.668686881 +0100
+@@ -2,3 +2,3 @@
+ ploplop
+-bloblob
++noooooo
+### <pkg:patches.1>
+opam-version: "2.0"
+patches: [
+  "to-apply.patch" { ?smtg }
+  "to-not-apply.patch" { !?smtg }
+]
+build: [
+  [ "grep" "ploplop" "some-content" ]
+  [ "grep" "bloblob" "some-content" ]
+]
+### opam install patches -vv | sed-cmd patch | 'patch-[^"]+' -> 'patch' | sed-cmd grep | '^- GNU patch .*' -> '- GNU patch' | unordered
+The following actions will be performed:
+=== install 1 package
+  - install patches 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+[patches: patch] applying to-apply.patch
++ patch "--version"
+- GNU patch
+- Copyright (C) 2003, 2009-2012 Free Software Foundation, Inc.
+- Copyright (C) 1988 Larry Wall
+- 
+- License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
+- This is free software: you are free to change and redistribute it.
+- There is NO WARRANTY, to the extent permitted by law.
+- 
+- Written by Larry Wall and Paul Eggert
```

Comment:
Is that chunk always the same cross platform? At least on ubuntu, macos & cygwin from CI run

Reply:
it looks like yeah